datafusion/spark/src/function/datetime/date_add.rs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::any::Any;
+use std::sync::Arc;
+
+use arrow::array::ArrayRef;
+use arrow::compute;
+use arrow::datatypes::{DataType, Date32Type};
+use datafusion_common::cast::{as_date32_array, as_int32_array};
+use datafusion_common::{internal_err, Result};
+use datafusion_expr::{
+    ColumnarValue, ScalarFunctionArgs, ScalarUDFImpl, Signature, Volatility,
+};
+use datafusion_functions::utils::make_scalar_function;
+
+#[derive(Debug)]
+pub struct SparkDateAdd {
+    signature: Signature,
+    aliases: Vec<String>,
+}
+
+impl Default for SparkDateAdd {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl SparkDateAdd {
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::exact(
+                vec![DataType::Date32, DataType::Int32],
```

Comment:
Spark accepts Int8, Int16, and Int32 for the second argument

```scala
  override def inputTypes: Seq[AbstractDataType] =
    Seq(DateType, TypeCollection(IntegerType, ShortType, ByteType))
```

In the Comet implementation of date_add/date_sub, we support all three as well:

https://github.com/apache/datafusion-comet/blob/main/native/spark-expr/src/datetime_funcs/date_arithmetic.rs#L53